src/web/cache.rs

```diff
@@ -415,6 +363,60 @@ mod tests {
             surrogate_control.map(|s| HeaderValue::from_str(s).unwrap())
         );
 
+        if needs_global_surrogate_key {
+            assert_eq!(headers.surrogate_keys.unwrap(), SURROGATE_KEY_ALL.into());
+        } else {
+            assert!(headers.surrogate_keys.is_none());
+        }
+
+        Ok(())
+    }
+
+    #[test]
+    fn render_fastly_forever_in_cdn() -> Result<()> {
```

Comment:
You'll need to add comments on these tests because even though the name is explicit, I think some extra explanations won't hurt.